zip-XXX-light-payment-detection.rst

```diff
@@ -0,0 +1,612 @@
+| 32    | nullifier    | char[32]  |
++-------+--------------+-----------+
+| 32    | rk           | char[32]  |
++-------+--------------+-----------+
+| 192   | zkproof      | char[192] |
++-------+--------------+-----------+
+| 64    | spendAuthSig | char[64]  |
++-------+--------------+-----------+
+
+The only part necessary for detection is the nullifier, which allows a light client to
+detect when one of its own notes has been spent. This means we only need to take 32 bytes
+of each Spend, for a 90% improvement in bandwidth use.
+
+Proxy operation
+===============
+
+The proxy's purpose is to provide a scalable and bandwidth-efficient interface between a
+Zcash node and any number of light clients. It accomplishes this by parsing a blockwise
+stream of transactions from the node and converting them into the compact format described
+above.
+
+The proxy offers the following API to clients:
+
+.. code:: proto
+
+    service CompactTxStreamer {
+        rpc GetLatestBlock(ChainSpec) returns (BlockID) {}
+        rpc GetBlock(BlockID) returns (CompactBlock) {}
+        rpc GetBlockRange(RangeFilter) returns (stream CompactBlock) {}
+        rpc GetTransaction(TxFilter) returns (FullTransaction) {}
+    }
+
+    // Remember that proto3 fields are all optional.
+
+    // Someday we may want to specify e.g. a particular chain fork.
+    message ChainSpec {}
+
+
+    // A BlockID message contains identifiers to select a block: either a
+    // height or a hash.
+    message BlockID {
+        uint64 blockHeight = 1;
+        bytes blockHash = 2;
+    }
+
+
+    message RangeFilter {
+        BlockID start = 1;
+        BlockID end = 2;
+    }
+
+    // A TxFilter contains the information needed to identify a particular
+    // transaction: either a block and an index, or a direct transaction hash.
+    message TxFilter {
+        BlockID blockID = 1;
+        uint64 txIndex = 2;
+        bytes txHash = 3;
+    }
+
+
+Client operation
+================
+
+Light clients obtain compact blocks from one or more proxy servers, which they then
+process locally to update their view of the block chain. We consider only a single proxy
+server here without loss of generality.
+
+Local processing
+----------------
+
+Given a ``CompactBlock`` received in height-sequential order from a proxy server, a light
+client can process it in four ways:
+
+Scanning for relevant transactions
+``````````````````````````````````
+For every ``CompactOutput`` in the ``CompactBlock``, the light client can trial-decrypt it
+against a set of Sapling incoming viewing keys. The procedure for trial-decrypting a
+``CompactOutput`` (*cmu*, *epk*, *ciphertext*) with an incoming viewing key *ivk* is a
+slight deviation from the standard decryption process [#sapling-ivk-decryption]_:
+
+- let sharedSecret = KA\ :sup:`Sapling`\ .Agree(*ivk*, *epk*)
+- let K\ :sup:`enc` = KDF\ :sup:`Sapling`\ (sharedSecret, *epk*)
+- let P\ :sup:`enc` = ChaCha20.Decrypt\ :sub:`K^enc`\ (*ciphertext*)
```

Comment:
This is not specified precisely enough. The intent is to perform the decryption as it would be performed by IETF_AEAD_CHACHA20_POLY1305, but this involves decrypting with the block counter starting from 1. Since ChaCha20 is not defined at all, it's not clear that the block counter should start with 1.

I suggest adding a definition like this:

```
Let ChaCha20.Decrypt\ :sub:`K^enc`\ (*initialCounter*, *ciphertext*) be the
decryption of *ciphertext* with key K^enc, initial counter value *initialCounter*,
and all-zero nonce, as specified in section 2.4 of [#RFC7539]_.
```
and then use ChaCha20.Decrypt<sub>K^enc</sub>(1, *ciphertext*).